receiver/redisreceiver/redis_metrics.go

```diff
@@ -0,0 +1,157 @@
+package redisreceiver
+
+import (
+	"errors"
+
+	metricsProto "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+)
+
+type redisMetric struct {
+	key               string
+	units             string
+	desc              string
+	labels            map[string]string
+	labelDescriptions map[string]string
+	metricType        metricType
+}
+
+func getDefaultRedisMetrics() []*redisMetric {
+	return []*redisMetric{
+		uptimeInSeconds(),
+		uptimeInDays(),
+
+		usedCpuSys(),
+		usedCpuSysChildren(),
+		usedCpuSysUserChildren(),
+
+		lruClock(),
+
+		connectedClients(),
+
+		clientRecentMaxInputBuffer(),
+		clientRecentMaxOutputBuffer(),
+
+		blockedClients(),
+
+		expiredKeys(),
+		evictedKeys(),
+
+		rejectedConnections(),
+
+		usedMemory(),
+		usedMemoryRss(),
+		usedMemoryPeak(),
+		usedMemoryLua(),
+
+		memFragmentationRatio(),
+
+		changesSinceLastSave(),
+
+		instantaneousOpsPerSec(),
+
+		rdbBgsaveInProgress(),
+
+		totalConnectionsReceived(),
+		totalCommandsProcessed(),
+
+		totalNetInputBytes(),
+		totalNetOutputBytes(),
+
+		keyspaceHits(),
+		keyspaceMisses(),
+
+		latestForkUsec(),
+
+		connectedSlaves(),
+
+		replBacklogFirstByteOffset(),
+
+		masterReplOffset(),
+	}
+}
+
+type metricType int
+
+const (
+	unspecified = iota
+	gaugeInt
+	gaugeDouble
+	gaugeDistribution
+	cumulativeInt
+	cumulativeDouble
+)
+
+func buildProtoMetrics(
+	redisInfo map[string]string,
+	redisMetrics []*redisMetric,
+) ([]*metricsProto.Metric, []error) {
+	var parsingErrors []error
+	var protoMetrics []*metricsProto.Metric
+	for _, redisMetric := range redisMetrics {
+		strVal := redisInfo[redisMetric.key]
+		if len(strVal) == 0 {
+			continue
+		}
+		protoMetric, parsingError := buildSingleProtoMetric(strVal, redisMetric)
+		if parsingError != nil {
+			parsingErrors = append(parsingErrors, parsingError)
+			continue
+		}
+		protoMetrics = append(protoMetrics, protoMetric)
+	}
+	return protoMetrics, parsingErrors
+}
+
+func buildSingleProtoMetric(strVal string, redisMetric *redisMetric) (*metricsProto.Metric, error) {
+	f := getPointConverter(redisMetric)
+	if f == nil {
+		return nil, errors.New("oops") // todo constant?
```

Comment:
Presumably this can never happen because we don't have redisMetrics of any type that is not an int or double. Can we somehow ensure that if we do we can a compile error or we have a test that ensures this is not possible? This would prevent future problems if someone adds a redisMetric with a different type to getDefaultRedisMetrics.